### ISSUE TITLE
Transitions need multi-methods, too.

### ISSUE THREAD

**Issue by 1wheel** [d3/d3-selection-multi#7]
`styles` after a transition doesn't work. Changing `index.js` to update the transition prototype fixes the problem:  

```
import {selection} from "d3-selection";
import selection_attrs from "./src/selection_attrs";
import selection_styles from "./src/selection_styles";
import selection_properties from "./src/selection_properties";

selection.prototype.attrs = selection_attrs;
selection.prototype.styles = selection_styles;
selection.prototype.properties = selection_properties;

if (selection.prototype.transition){
  selection.prototype.transition.attrs = selection_attrs;
  selection.prototype.transition.styles = selection_attrs;
  selection.prototype.transition.properties = selection_attrs;
}
```

But it seems a bit hacky since it there's an unspecified dependency on loading `d3-transition` before `d3-selection-multi`. I'm not sure if there should be a separate `d3-selection-transition-multi` module to handle this? Creating two modules for everything you'd like to stick onto `selection.prototype` seems like overkill.

**Comment by mbostock** [comment]
Right, this module only defines the multi-methods for selections, not transitions.

You can’t use the methods in this module on transitions. It’ll work (coincidentally) if you pass an object to _transition_.attrs, for example, but it won’t work if you pass a function that returns an object. See [the implementation](https://github.com/d3/d3-selection-multi/blob/master/src/selection_attrs.js). The latter is a new feature; it works like this:

``` js
selection.attrs(function(d) { return {foo: d.foo}; });
```

Which is equivalent to:

``` js
selection.attrs({foo: function(d) { return d.foo; }});
```

The advantage is that you can share some per-element computation across attributes and also define which attributes are set per-element, say using a dynamic property name in ES2015:

``` js
selections.attrs((d) => ({["data-" + d.key]: d.value}));
```

If _d_ is `{key: "foo", value: 42}`, this would set the “data-foo” attribute to 42.

Given that a separate implementation is required, I guess I’d lean towards a new d3-transition-multi module. Alternatively, I would add a hard dependency on d3-transition to this module, so that it works reliably (rather than failing silently if d3-transition is loaded after d3-selection-multi).

**Comment by mbostock** [comment]
Fixed in 0.4.
